src/components/UI/common/NewsletterSignup.tsx

```diff
@@ -1,16 +1,60 @@
-import { Flex, Input, Stack } from '@chakra-ui/react';
+import { Button, Center, Flex, FormControl, Input, Stack, useToast } from '@chakra-ui/react';
 import { FC } from 'react';
+import { useForm } from 'react-hook-form';
 
-import { ImportantText, PageText } from '../text';
+import { ImportantText } from '../headings';
+import { PageText } from '../text';
+
+type FormData = {
+  email: string;
+};
 
 export const NewsletterSignup: FC = () => {
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isValid },
+    reset
+  } = useForm<FormData>({
+    mode: 'onChange'
+  });
+  const toast = useToast();
+
+  const onSubmit = (data: FormData) => {
+    if (errors.email) {
```

Comment:
Curious, would this ever be different from `isValid`?

Reply:
It's an edge case, not sure if it makes sense. `isValid` disables the submit button while `errors` containts info about the input no passing the validations. You can't submit and invalid email directly because of the disabled submit, so maybe it can be removed.